File: rzops-web/src/lib/types/credential.ts
export interface CredentialResponse {
  id: string;
  name: string;
  credential_type: string;
  username: string | null;
  secret_ref: string | null;
  owner_id: string | null;
  status: string;
  remarks: string | null;
  created_at: string;
  updated_at: string;
}

export interface CredentialListResponse {
  data: CredentialResponse[];
  count: number;
}

export interface CreateCredentialRequest {
  name: string;
  credential_type: string;
  username?: string;
  secret_ref?: string;
  owner_id?: string;
  status?: string;
  remarks?: string;
}

export interface UpdateCredentialRequest extends Partial<CreateCredentialRequest> {}

export interface ListCredentialsQuery {
  q?: string;
  status?: string;
  credential_type?: string;
  limit?: number;
  offset?: number;
}

File: rzops-api/domain/src/models/credential.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;


/// Credential entity — maps to cmdb_credential.
#[derive(Debug, Clone)]
pub struct Credential {
    pub id: Uuid,
    pub name: String,
    pub credential_type: String,
    pub username: Option<String>,
    pub secret_ref: Option<String>,
    pub owner_id: Option<Uuid>,
    pub status: String,
    pub remarks: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
